tests/core/e2e/test_e2e_finetune_hydra.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import annotations
+
+import os
+import tempfile
+from pathlib import Path
+
+import pytest
+import torch
+from test_e2e_commons import _run_main, oc20_lmdb_train_and_val_from_paths
+
+from fairchem.core.models.finetune_hydra import FTHYDRA_NAME, FineTuneMode, FTConfig
+
+
+@pytest.fixture()
+def tutorial_val_src(tutorial_dataset_path):
+    return tutorial_dataset_path / "s2ef/val_20"
+
+
+def make_checkpoint(data_source, seed) -> str:
+    # first train a tiny eqv2 model to get a checkpoint
+    eqv2_yml = Path("tests/core/models/test_configs/test_equiformerv2_hydra.yml")
+    tempdir = tempfile.TemporaryDirectory()
+    ck_path = os.path.join(tempdir.name, "checkpoint.pt")
+    _run_main(
+        tempdir.name,
+        eqv2_yml,
+        update_dict_with={
+            "optim": {
+                "max_epochs": 1,
+                "eval_every": 8,
+                "batch_size": 1,
+                "num_workers": 0,
+            },
+            "dataset": oc20_lmdb_train_and_val_from_paths(
+                train_src=str(data_source),
+                val_src=str(data_source),
+                test_src=str(data_source),
+                otf_norms=False,
+            ),
+        },
+        update_run_args_with={"seed": seed},
+        save_checkpoint_to=ck_path,
+        world_size=0,
+    )
+    assert os.path.isfile(ck_path)
+    return ck_path, tempdir
+
+
+def run_main_with_ft_hydra(tempdir: tempfile.TemporaryDirectory,
+                           yaml: str,
+                           data_src: str,
+                           run_args: dict,
+                           ft_config: str,
+                           output_checkpoint: str):
+    _run_main(
+        tempdir.name,
+        yaml,
+        update_dict_with={
+            "optim": {
+                "max_epochs": 1,
+                "eval_every": 8,
+                "batch_size": 1,
+                "num_workers": 0,
+                "lr_initial": 0.0 # don't learn anything
+            },
+            "dataset": oc20_lmdb_train_and_val_from_paths(
+                train_src=str(data_src),
+                val_src=str(data_src),
+                test_src=str(data_src),
+                otf_norms=False,
+            ),
+            "model": {
+                "name": FTHYDRA_NAME,
+                FTConfig.FT_CONFIG_NAME: ft_config,
+            }
+        },
+        update_run_args_with=run_args,
+        save_checkpoint_to=output_checkpoint,
+        world_size=0,
+    )
+
+
+def test_finetune_hydra_retain_backbone(tutorial_val_src):
+    starting_ckpt, original_tmpdir = make_checkpoint(tutorial_val_src, 0)
+    old_state_dict = torch.load(starting_ckpt)["state_dict"]
+    # now finetune a the model with the checkpoint from the first job
+    tempdir2 = tempfile.TemporaryDirectory()
```

Comment:
Does this temporarydirectory and the one inside make_checkpoint need to be inside a try/finally or a with in case the test fails with an error and it is not cleaned up?

Reply:
fixed, changed all to with statements